response_operations_ui/views/accounts.py

```diff
@@ -94,6 +95,53 @@ def change_account_name():
     return render_template("account/change-account-name.html", user=user, form=form, errors=form.errors)
 
 
+@account_bp.route("/change-username", methods=["GET", "POST"])
+@login_required
+def change_username():
+    form = UsernameChangeForm()
+    user_id = session["user_id"]
+    user_from_uaa = uaa_controller.get_user_by_id(user_id)
+    username = user_from_uaa["userName"]
+    username_exists = False
+    if request.method == "POST" and form.validate():
+        user_from_uaa["userName"] = form["username"].data
```

Comment:
put this inside the if statement below.  No point in doing this assignment until you've checked that they're not the same